x-pack/plugins/ml/public/explorer/explorer_react_wrapper_directive.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * AngularJS directive wrapper for rendering Anomaly Explorer's React component.
+ */
+
+import _ from 'lodash';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { Explorer } from './explorer';
+
+import { IntervalHelperProvider } from 'plugins/ml/util/ml_time_buckets';
+import { SWIMLANE_TYPE } from './explorer_constants';
+
+import { uiModules } from 'ui/modules';
+const module = uiModules.get('apps/ml');
+
+import { I18nProvider } from '@kbn/i18n/react';
+
+module.directive('mlExplorerReactWrapper', function (Private) {
+  const TimeBuckets = Private(IntervalHelperProvider);
+
+  function link(scope, element) {
+    function getSwimlaneData(swimlaneType) {
+      switch (swimlaneType) {
+        case SWIMLANE_TYPE.OVERALL:
+          return scope.overallSwimlaneData;
+        case SWIMLANE_TYPE.VIEW_BY:
+          return scope.viewBySwimlaneData;
+      }
+    }
+
+    function mapScopeToSwimlaneProps(swimlaneType) {
+      return {
+        chartWidth: scope.swimlaneWidth,
+        MlTimeBuckets: TimeBuckets,
+        swimlaneCellClick: scope.swimlaneCellClick,
+        swimlaneData: getSwimlaneData(swimlaneType),
+        swimlaneType,
+        selection: scope.appState.mlExplorerSwimlane,
+      };
+    }
+
+    function render() {
+      const props = _.pick(scope, [
```

Comment:
as only `pick` is needed, you could only import that function from `lodash`

Reply:
Done in 85d1567592.